### DISCUSSION TITLE
[AI-Deck] Classification Demo cfloader flashing hangs at 99%

### DISCUSSION THREAD

**Discussion by chengliu-LR** [bitcraze/discussions#315] [Q&A]
Hi all,

I am trying to flash the classification demo to ai-deck through `cfloader`. However, the flashing hangs at `\ 99% Writing to bcAI:gap8 deck memory` everytime. The same flashing process goes totally well for the WiFi Video Streamer demo. I am working with crazyflie 2.X and ai-deck v1.1, BTW. Is there any clue to solve this problem?

**Comment by evoggy** [comment]
Hi,

Not sure what the issue could be, but it might have something to do with the size. Did you make any changes to the code in between the tests for flashing it (i.e did the size change)? Could you run [this script](https://github.com/bitcraze/aideck-gap8-bootloader/blob/master/check-app-image.py) on the resulting binary and paste the output?

**Reply by chengliu-LR** [reply]
Hi Marcus,

Thank you so much for your reply! For your concern, I followed the instructions in the [classification demo](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/ai-examples/classification-demo/), without retraining the model and ran this command in the `gap_sdk` container directly: `$ docker run --rm -v ${PWD}:/module aideck-with-autotiler tools/build/make-example examples/ai/classification clean model build image`. I got the `target.board.devices.flash.img` image that can be flashed through `cfloader`, and thats where my issue came in. I also ran your suggested script for the resulting binary `target.board.devices.flash.img` and here is what I got:

```
> python3 check-app-image.py target.board.devices.flash.img                                                                                   (ai-deck) 10/20/22 -  1:40 AM
Showing info for target.board.devices.flash.img
Size on disk: 747792
Size: 200704
nSegments: 5
Entrypoint: 0x1C000094
Entrypoint base?: 0x1C000000

[0] Flash offset = 0x400        RAM offset = 0x1B000000 size = 0xC      nBlocks = 1
[1] Flash offset = 0x800        RAM offset = 0x1B00000C size = 0x58C    nBlocks = 2
[2] Flash offset = 0x1000       RAM offset = 0x1C000000 size = 0x2F78C  nBlocks = 190
[3] Flash offset = 0x30800      RAM offset = 0x1C0318F0 size = 0x78     nBlocks = 1
[4] Flash offset = 0x30C00      RAM offset = 0x1C031968 size = 0x8      nBlocks = 1

L1 size to copy: 0x598 (1432)
L2 size to copy: 0x2F80C (194572)

No overlap with bootloader, its all fine!
```

**Reply by evoggy** [reply]
Apologies for the late reply on this one! Looking at the output of the script it seems fine and I can't see the issue I was suspecting. Any chance you could share the binary? Or is it just a clean build of the demo?
